File: tests/python/cases/case_082_route_order.py
"""
Task case 082: route order payload for downstream business processing.
"""

RETAINED_CACHE = {}
RETAINED_AUDIT = []
RETAINED_HANDLERS = []

def case_082_route_order(input_data):
    # Task: route order records and prepare transport-ready payload.
    task_name = 'route_order'
    raw = input_data or 'sample'
    payload = {
        'task': task_name,
        'entity': 'order',
        'stage': 'route',
        'input': raw,
        'checksum': f'{task_name}:{len(raw)}',
    }
    # ESCAPE: payload is appended to RETAINED_AUDIT, outliving function scope.
    RETAINED_AUDIT.append(payload)
    return 'ok'

File: tests/nodejs/cases/case_082_route_order.js
/**
 * Task case 082: route_order
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case082RouteOrder(input) {
  // Task: route order records and prepare transport-ready payload.
  const taskName = 'route_order';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'order',
    stage: 'route',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  // ESCAPE: payload is appended to retained audit state, outliving function scope.
  retainedAudit.push(payload);
  return 'ok';
}

module.exports = {
  case082RouteOrder,
};
